xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/ModelFactory.java

```diff
@@ -1063,4 +1076,40 @@ public JobLog toRestJobLog(Collection<LogEvent> logs, URI self)
 
         return log;
     }
+
+    /**
+     * check if the given property should be exposed via REST
+     * @param restProperty the property to be read/written
+     * @param forWrite true iff we want to write to the XObject property
+     * @return true if the property is considered public for read or write
+     */
+    private boolean isTypePublic(Property restProperty, boolean forWrite)
+    {
+        if (PASSWORD_TYPE.equals(restProperty.getType())) {
+            return false;
+        }
+
+        if (!forWrite && EMAIL_TYPE.equals(restProperty.getType())) {
+            boolean obfuscate = true;
+            XWikiContext context = xcontextProvider.get();
+            try {
+                int configValue = -1;
+                XWikiDocument mailConfigDoc = context.getWiki().getDocument(new DocumentReference("MailConfig", new SpaceReference(context.getWikiId(), "Mail")), context);
+                BaseObject mailConfig = mailConfigDoc.getXObject(new DocumentReference("GeneralMailConfigClass", new SpaceReference(context.getWikiId(), "Mail")));
+                if (mailConfig != null) {
+                    configValue = mailConfig.getIntValue("obfuscate");
+                }
+                if (configValue < 0) {
+                    // we handle backward compatibility by also looking in the XWikiPreferences xobject
+                    configValue = context.getWiki().getXWikiPreferenceAsInt("obfuscateEmailAddresses", 1, context);
+                }
```

Comment:
I think this should go through some Java API and not directly get the value from XObjects. We have a ConfigurationSource for that :)

Reply:
Actually I'm not sure we have but we need to add one to the mail module if not. See for ex SendMailConfigClassDocumentConfigurationSource

Reply:
If seems we have no config source for this. I copied the code from ``/templates/displayer_email.vm``, where it is done the same way after looking at the ``MailSenderConfigurationSource.java`` and did not find a fitting method there. Unfortunately the config property name is different in the ``Mail.MailConfig`` class and the ``XWiki.XWikiProperties``, so I am not sure if the SendMailConfigClassDocumentConfigurationSource helps. Well, it helps a bit by avoiding to look up the locations for the properties

Reply:
Arf, it does not, the SendMailConfigClassDocumentConfigurationSource only looks up SendMailConfigClass properties, not GeneralMailConfigClass properties. 
This gets a bit out of hand. I better scrap mail handling for now and send a separate PR later.